Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/sparkserverlessnode/SparkServerlessClusterNode.java

```diff
@@ -90,7 +90,7 @@ protected void loadActions() {
                 this, cluster, adlAccount, SparkServerlessClusterOps.getInstance().getDestroyAction()));
         addAction("View Cluster Status", new SparkServerlessMonitorAction(
                 this, cluster, SparkServerlessClusterOps.getInstance().getMonitorAction()));
-        if (Optional.ofNullable(cluster.getMasterState()).orElse("")
+        if (getNodeActionByName("Update") == null && Optional.ofNullable(cluster.getMasterState()).orElse("")
                 .equals(SparkItemGroupState.STABLE.toString())) {
             addAction("Update", new SparkServerlessUpdateAction(
```

Comment:
why the action needs to be added twice?

Reply:
After user provision a new cluster, the cluster state is not `STABLE`. So the actions for this new provisioned cluster node don't contain `Update`.  `Update` action will be enabled when cluster is in `STABLE` state and user refreshes the node.

---
In reply to: [217964570](https://github.com/Microsoft/azure-tools-for-java/pull/2013#discussion_r217964570) [](ancestors = 217964570)